### PR TITLE
[RNMobile] another approach to fix jumping toolbar

### LINKED ISSUE #2301: Button: Toolbar goes up and down for a split second when dismissing settings
**To Reproduce**
1. Add a Button
2. Click on Button block settings
3. Focus on URL field
4. Close the Bottom sheet by tapping outside of it

Result: Toolbar goes up and down detaching from the keyboard for a split second

**Expected behavior**
Toolbar stays on the keyboard

**Screenshots**

May-22-2020 12-48-08

**Smartphone (please complete the following information):**
 - Device: iPhoneXS
 - OS: iOS 13
 - Version WordPress iOS 14.8

### PR DESCRIPTION
**Pull request by lukewalczak**
## Description

Fix jumping toolbar after closing the `LinkSettings` when Button/Paragraph is focused

Fixes: https://github.com/wordpress-mobile/gutenberg-mobile/issues/2301

### TEST CASE 1
1. Add a Button
2. Click on Button block settings
3. Focus on URL field
4. Close the Bottom sheet by tapping outside of it
**5. Expect: toolbar mustn't jump**

### TEST CASE 2
1. Open mobile app
2. Press FAB 
3. Choose `Site page`
**4. Expect to see layout template buttons**
5. Press paragraph
6. Press link icon
7. Close bottom sheet
**8. Expect: toolbar mustn't jump**

### TEST CASE 3
1. Open mobile app
2. Open new post/site
3. Press `...` button in the right corner
4. Choose `Switch to HTML mode`
5. Paste the HTML code

<details>

```
<!-- wp:paragraph -->
<p></p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p></p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p></p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p></p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p></p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p></p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p></p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p></p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p></p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p></p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p></p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p></p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p></p>
<!-- /wp:paragraph -->
```

</details>

**6. Expect to be able to scroll the content**
7. Focus one of the paragraphs
**8. Expect to be able to edit the content and scroll it** 

Test as well [Buttons-1-2-3](https://github.com/wordpress-mobile/test-cases/blob/master/test-suites/gutenberg/sanity-tests.md)

## Screenshots 

iphone 8 | iphone 11
--- | ---
jump_toolbar_ok | keyboard_iX_2

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by chipsnyder — APPROVED** [review]
Tested and it's working as described for me. Thanks for the quick fix @lukewalczak!

File: packages/components/src/mobile/keyboard-avoiding-view/index.ios.js [hunk 1]
```diff
@@ -3,19 +3,103 @@
...
+	const paddedKeyboardHeight =
+		keyboardHeight + MIN_HEIGHT - ( style.bottom || 0 );
+
+	function animate() {
+		if ( keyboardHeight ) {
+			animatedHeight.setValue( paddedKeyboardHeight );
+		} else
+			Animated.timing( animatedHeight, {
+				toValue: MIN_HEIGHT,
+				duration: 150,
```
Comment (dratwas) [inline packages/components/src/mobile/keyboard-avoiding-view/index.ios.js:74]: Just wondering why it is `150`? :)

File: packages/block-library/src/button/edit.native.js [hunk 2]
```diff
@@ -372,7 +372,6 @@ class ButtonEdit extends Component {
 						__unstableMobileNoFocusOnMount={ ! isSelected }
 						selectionColor={ textColor }
 						onBlur={ () => {
-							this.onToggleButtonFocus( false );
```
Comment (dratwas) [inline packages/block-library/src/button/edit.native.js:375]: Could you explain to me why this change is needed?

Comment (lukewalczak) [reply packages/block-library/src/button/edit.native.js:375]: I found this line of code redundant because it's introducing unnecessary state change - button is still selected and after closing the bottom sheet `RichText` is focused so there is no need to change that local state value. Additionally, styling buttons on the toolbar above the keyboard was disappearing:

before | after
--- | ---
 | ok

File: packages/components/src/mobile/keyboard-avoiding-view/index.ios.js (hunk 1 above)
Comment (lukewalczak) [reply packages/components/src/mobile/keyboard-avoiding-view/index.ios.js:74]: When I tested it before, this value seemed to be correct to me, however, I have decided to set the value to `200` and compare it with the hiding time of the keyboard and have to say that `200` is the better choice.

File: packages/block-library/src/button/edit.native.js (hunk 2 above)
Comment (dratwas) [reply packages/block-library/src/button/edit.native.js:375]: Thanks for the details!

File: packages/components/src/mobile/keyboard-avoiding-view/index.ios.js (hunk 1 above)
Comment (dratwas) [reply packages/components/src/mobile/keyboard-avoiding-view/index.ios.js]: I believe you could take a `duration` from Keyboard event :)

Comment (lukewalczak) [reply packages/components/src/mobile/keyboard-avoiding-view/index.ios.js]: Done - passed duration from Keyboard event!

File: packages/components/src/mobile/keyboard-avoiding-view/index.ios.js [hunk 3]
```diff
@@ -3,19 +3,99 @@
...
+					result.safeAreaInsets.bottom
+			);
+		} );
+	}
+
+	function onKeyboardWillHide( { duration } ) {
+		animate( duration );
+	}
+
+	function animate( duration ) {
```
Comment (dratwas) [inline packages/components/src/mobile/keyboard-avoiding-view/index.ios.js:67]: I am wondering if we really need the animate function. I see that it's used only once in the `onKeyboardWillHide`. Wdyt?

**Review by dratwas — APPROVED** [review]
